File: app/src/main/java/info/upump/jym/entity/Day.java
package info.upump.jym.entity;

import info.upump.jym.R;


public enum Day {
    MONDAY(){
        @Override
        public int getColor() {
            return R.color.monday ;
        }

        @Override
        public int getName() {
            return R.string.day_monday;
        }
    },
    TUESDAY(){
        @Override
        public int getColor() {
            return R.color.tuesday ;
        }

        @Override
        public int getName() {
            return R.string.day_tuesday;
        }
    },
    WEDNESDAY (){
        @Override
        public int getColor() {
            return R.color.wednesday ;
        }

        @Override
        public int getName() {
            return R.string.day_wednesday;
        }
    },
    THURSDAY(){
        @Override
        public int getColor() {
            return R.color.thursday;
        }

        @Override
        public int getName() {
            return R.string.day_thursday;
        }
    },
    FRIDAY(){
        @Override
        public int getColor() {
            return R.color.friday ;
        }

        @Override
        public int getName() {
            return R.string.day_friday;
        }
    },
    SATURDAY(){
        @Override
        public int getColor() {
            return R.color.saturday ;
        }

        @Override
        public int getName() {
            return R.string.day_saturday;
        }
    },
    SUNDAY(){
        @Override
        public int getColor() {
            return R.color.sunday;
        }

        @Override
        public int getName() {
            return R.string.day_sunday;
        }
    };

    public abstract int getColor();
    public abstract int getName();
}

File: mycompose/src/main/java/info/upump/mycompose/models/entity/Day.kt
package info.upump.mycompose.models.entity

import info.upump.mycompose.R


enum class Day {
    MONDAY {
        override fun getColor(): Int {
            return R.color.monday
        }

        override fun getName(): Int {
            return R.string.day_monday
        }
    },
    TUESDAY {
        override fun getColor(): Int {
            return R.color.tuesday
        }

        override fun getName(): Int {
            return R.string.day_tuesday
        }
    },
    WEDNESDAY {
        override fun getColor(): Int {
            return R.color.wednesday
        }

        override fun getName(): Int {
            return R.string.day_wednesday
        }
    },
    THURSDAY {
        override fun getColor(): Int {
            return R.color.thursday
        }

        override fun getName(): Int {
            return R.string.day_thursday
        }
    },
    FRIDAY {
        override fun getColor(): Int {
            return R.color.friday
        }

        override fun getName(): Int {
            return R.string.day_friday
        }
    },
    SATURDAY {
        override fun getColor(): Int {
            return R.color.saturday
        }

        override fun getName(): Int {
            return R.string.day_saturday
        }
    },
    SUNDAY {
        override fun getColor(): Int {
            return R.color.sunday
        }

        override fun getName(): Int {
            return R.string.day_sunday
        }
    };

    abstract fun getColor(): Int
    abstract fun getName(): Int
}